hello-wiredcraft/src/controllers/user.controller.ts

```diff
@@ -0,0 +1,135 @@
+import {
+  authenticate,
+  TokenService,
+  UserService,
+} from '@loopback/authentication';
+import {inject} from '@loopback/core';
+import {repository} from '@loopback/repository';
+import {
+  del,
+  get,
+  HttpErrors,
+  param,
+  post,
+  put,
+  requestBody,
+} from '@loopback/rest';
+import {SecurityBindings, securityId, UserProfile} from '@loopback/security';
+import _ from 'lodash';
+import {NewUser, User} from '../models';
+import {Credentials, UserRepository} from '../repositories';
+import {TokenServiceBindings, UserServiceBindings} from '../services';
+import {
+  CREATE_USER_REQUEST_SPEC,
+  CREATE_USER_RESPONSE_SPEC,
+  DELETE_USER_RESPONSE_SPEC,
+  GET_USER_ME_RESPONSE_SPEC,
+  GET_USER_RESPONSE_SPEC,
+  UPDATE_USER_REQUEST_SPEC,
+  UPDATE_USER_RESPONSE_SPEC,
+  USER_LOGIN_REQUEST_SPEC,
+  USER_LOGIN_RESPONSE_SPEC,
+} from '../specs/user-spec';
+
+export class UserController {
+  constructor(
+    @repository(UserRepository)
+    public userRepository: UserRepository,
+    @inject(TokenServiceBindings.TOKEN_SERVICE) public jwtService: TokenService,
+    @inject(UserServiceBindings.USER_SERVICE)
+    public userService: UserService<User, Credentials>,
+  ) {}
+
+  @post('/users', CREATE_USER_RESPONSE_SPEC)
+  async create(
+    @requestBody(CREATE_USER_REQUEST_SPEC)
+    newUser: NewUser,
+  ): Promise<User> {
+    const existUser = await this.userRepository.findOne({
+      where: {name: newUser.name},
+    });
+    if (this.isExist(existUser)) {
+      throw new HttpErrors.Conflict('The user is already exists');
```

Comment:
Good comment.
I will add a try-catch for it.
When catch error, if the `existUser` is not null, I will mark the user `deleted=true`.
Otherwise, I will delete the `user` and the `userCredentials`.

Reply:
From my point of view: 
First, I will update the error message to 'Create user fail'.
Secondly, add a captcha.
Finally,  add a rate limit in the Loadbalancer or APIGateway?

Reply:
thanks, it's good enough to have answers from you, I didn't ask for enhancement :) @Tong-Huang